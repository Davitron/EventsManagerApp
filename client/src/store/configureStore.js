import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../reducers/rootReducer';

const configureStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

configureStore.subscribe(() => {
  console.log('STORE SUB...', configureStore.getState());
});

export default configureStore;
