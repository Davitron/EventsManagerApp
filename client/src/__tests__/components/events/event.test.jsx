import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedEvent, { Event } from '../../../components/event/events';

jest.mock('../../../helpers/image-upload', () => jest.fn(() => Promise.resolve({})));

let mountedComponent;

let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  },
};

const response = {
  centers: {},
  newCenter: {},
  allStates: {},
};

const dispatch = fn => fn;

const updateEvent = jest.fn();
let deleteId = null;
const deleteEvent = (id) => {
  deleteId = id;
};

const location = {};


const mockStore = configureStore();
let wrapper;

/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      dispatch,
      response,
      history,
      location,
      updateEvent,
      deleteEvent,
    };
    history.push = jest.fn();
    mountedComponent = shallow(<Event {...props} />);
  }
  return mountedComponent;
};

describe('Event Component', () => {
  beforeAll(() => {
    wrapper = getComponent();
  });


  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      response: {
        ...response,
        events: {
          status: 'success',
          data: {
            message: 'Events Retrieved',
            data: [
              { id: 1, eventName: 'new Event' },
              { id: 2, eventName: ' another Event' }
            ],
            metadata: {
              pagination: {
                page: 1,
                limit: 9
              }
            }
          }
        },
        newEvent: {
          status: 'success',
          data: {
            eventName: 'new Event',
          }
        },
        deleteStatus: {
          status: 'success',
        }
      },
      location: {
        search: '/events'
      }
    };

    const eventsResponse = nextProps.response.events;
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().openModal).toEqual(false);
    expect(wrapper.state().isRequestMade).toEqual(false);
    expect(wrapper.state().data).toEqual(eventsResponse.data.data);
  });

  it('testing on change page function', () => {
    wrapper.instance().onChangePage(1);
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on page size change', () => {
    wrapper.instance().onPageSizeChange(1, 9);
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on search function', () => {
    wrapper.instance().onSearch({ limit: 9, page: 1 });
    expect(history.push).toHaveBeenCalled();
  });

  it('testing on onSubmit function', () => {
    const event = {
      eventName: 'Test Arena',
      centerId: 2,
      days: '2',
      image: {
        files: []
      },
      startDate: '2018-12-12'
    };
    wrapper = getComponent();
    wrapper.instance().onSubmit(event);
    expect(wrapper.state().openModal).toEqual(false);
    expect(wrapper.state().isRequestMade).toEqual(true);
  });

  it('testing on onSubmit function', () => {
    const event = {
      eventName: 'Test Arena',
      centerId: 2,
      days: 'laaa',
      image: {
        files: []
      },
      startDate: '2018-12-12'
    };

    wrapper.instance().onSubmit(event);
    expect(Object.keys(wrapper.state().errors).length).toBeGreaterThan(0);
    expect(wrapper.state().isRequestMade).toEqual(false);
  });

  it('testing show modal function', () => {
    const event = {
      eventName: 'Test Arena',
      centerId: 2,
      days: '2',
      image: {
        files: []
      },
      startDate: '2018-12-12'
    };

    wrapper.instance().showModal(event);
    expect(wrapper.state().errors).toEqual({});
    expect(wrapper.state().selectedEvent).toEqual(event);
    expect(wrapper.state().openModal).toEqual(true);
  });

  it('testing hide modal function', () => {
    wrapper.instance().hideModal();
    expect(wrapper.state().openModal).toEqual(false);
  });

  it('testing show modal function', () => {
    const event = {
      eventName: 'Test Arena',
      centerId: 2,
      days: '2',
      image: {
        files: []
      },
      startDate: '2018-12-12'
    };
    wrapper.instance().showPrompt(event);
    expect(wrapper.state().isRequestMade).toEqual(false);
    expect(wrapper.state().selectedEvent).toEqual(event);
    expect(wrapper.state().openPrompt).toEqual(true);
  });

  it('testing hide prompt function', () => {
    wrapper.instance().hidePrompt();
    expect(wrapper.state().openPrompt).toEqual(false);
  });

  it('testing handle delete function', () => {
    wrapper.setState({
      selectedEvent: {
        id: 2,
        eventName: 'test event'
      }
    });
    wrapper.instance().handleDelete();
    expect(deleteId).toBe(wrapper.state().selectedEvent.id);
  });
});

describe('Connected Component', () => {
  describe('Connected Events Component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        response,
      });
      wrapper = shallow(<ConnectedEvent store={store} history={history} location={location} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
