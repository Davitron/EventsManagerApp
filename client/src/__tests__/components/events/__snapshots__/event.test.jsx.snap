// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    deleteEvent={[Function]}
    dispatch={[Function]}
    getAll={[Function]}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/centers",
            ],
          ],
        },
      }
    }
    location={Object {}}
    response={
      Object {
        "allStates": Object {},
        "centers": Object {},
        "newCenter": Object {},
      }
    }
    updateEvent={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="background"
      >
        <div
          className="my-container"
        >
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <span
              style={
                Object {
                  "fontSize": "45px",
                }
              }
            >
              Events
            </span>
          </div>
          <Grid>
            <GridRow
              colunms={3}
            >
              <h2
                className="animated fadeInUp"
              >
                No Events Retrieved
              </h2>
            </GridRow>
          </Grid>
        </div>
        <Connect(EventFormModal)
          errors={Object {}}
          event={undefined}
          hideModal={[Function]}
          isRequestMade={false}
          onSubmit={[Function]}
          open={false}
        />
        <Prompt
          isRequestMade={false}
          message="Are you sure you want to delete this Event"
          onCancel={[Function]}
          onConfirm={[Function]}
          open={false}
          title="Delete Event"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="my-container"
          >
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontSize": "45px",
                  }
                }
              >
                Events
              </span>
            </div>
            <Grid>
              <GridRow
                colunms={3}
              >
                <h2
                  className="animated fadeInUp"
                >
                  No Events Retrieved
                </h2>
              </GridRow>
            </Grid>
          </div>,
          <Connect(EventFormModal)
            errors={Object {}}
            event={undefined}
            hideModal={[Function]}
            isRequestMade={false}
            onSubmit={[Function]}
            open={false}
          />,
          <Prompt
            isRequestMade={false}
            message="Are you sure you want to delete this Event"
            onCancel={[Function]}
            onConfirm={[Function]}
            open={false}
            title="Delete Event"
          />,
        ],
        "className": "background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Events
                </span>
              </div>,
              <Grid>
                <GridRow
                  colunms={3}
                >
                  <h2
                    className="animated fadeInUp"
                  >
                    No Events Retrieved
                  </h2>
                </GridRow>
              </Grid>,
              false,
            ],
            "className": "my-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Events
                </span>,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Events",
                  "style": Object {
                    "fontSize": "45px",
                  },
                },
                "ref": null,
                "rendered": "Events",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GridRow
                  colunms={3}
                >
                  <h2
                    className="animated fadeInUp"
                  >
                    No Events Retrieved
                  </h2>
                </GridRow>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    false,
                    <h2
                      className="animated fadeInUp"
                    >
                      No Events Retrieved
                    </h2>,
                    Array [],
                  ],
                  "colunms": 3,
                },
                "ref": null,
                "rendered": Array [
                  false,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No Events Retrieved",
                      "className": "animated fadeInUp",
                    },
                    "ref": null,
                    "rendered": "No Events Retrieved",
                    "type": "h2",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "errors": Object {},
            "event": undefined,
            "hideModal": [Function],
            "isRequestMade": false,
            "onSubmit": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isRequestMade": false,
            "message": "Are you sure you want to delete this Event",
            "onCancel": [Function],
            "onConfirm": [Function],
            "open": false,
            "title": "Delete Event",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="background"
        >
          <div
            className="my-container"
          >
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontSize": "45px",
                  }
                }
              >
                Events
              </span>
            </div>
            <Grid>
              <GridRow
                colunms={3}
              >
                <h2
                  className="animated fadeInUp"
                >
                  No Events Retrieved
                </h2>
              </GridRow>
            </Grid>
          </div>
          <Connect(EventFormModal)
            errors={Object {}}
            event={undefined}
            hideModal={[Function]}
            isRequestMade={false}
            onSubmit={[Function]}
            open={false}
          />
          <Prompt
            isRequestMade={false}
            message="Are you sure you want to delete this Event"
            onCancel={[Function]}
            onConfirm={[Function]}
            open={false}
            title="Delete Event"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="my-container"
            >
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "fontSize": "45px",
                    }
                  }
                >
                  Events
                </span>
              </div>
              <Grid>
                <GridRow
                  colunms={3}
                >
                  <h2
                    className="animated fadeInUp"
                  >
                    No Events Retrieved
                  </h2>
                </GridRow>
              </Grid>
            </div>,
            <Connect(EventFormModal)
              errors={Object {}}
              event={undefined}
              hideModal={[Function]}
              isRequestMade={false}
              onSubmit={[Function]}
              open={false}
            />,
            <Prompt
              isRequestMade={false}
              message="Are you sure you want to delete this Event"
              onCancel={[Function]}
              onConfirm={[Function]}
              open={false}
              title="Delete Event"
            />,
          ],
          "className": "background",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "fontSize": "45px",
                      }
                    }
                  >
                    Events
                  </span>
                </div>,
                <Grid>
                  <GridRow
                    colunms={3}
                  >
                    <h2
                      className="animated fadeInUp"
                    >
                      No Events Retrieved
                    </h2>
                  </GridRow>
                </Grid>,
                false,
              ],
              "className": "my-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    style={
                      Object {
                        "fontSize": "45px",
                      }
                    }
                  >
                    Events
                  </span>,
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Events",
                    "style": Object {
                      "fontSize": "45px",
                    },
                  },
                  "ref": null,
                  "rendered": "Events",
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <GridRow
                    colunms={3}
                  >
                    <h2
                      className="animated fadeInUp"
                    >
                      No Events Retrieved
                    </h2>
                  </GridRow>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      false,
                      <h2
                        className="animated fadeInUp"
                      >
                        No Events Retrieved
                      </h2>,
                      Array [],
                    ],
                    "colunms": 3,
                  },
                  "ref": null,
                  "rendered": Array [
                    false,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "No Events Retrieved",
                        "className": "animated fadeInUp",
                      },
                      "ref": null,
                      "rendered": "No Events Retrieved",
                      "type": "h2",
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errors": Object {},
              "event": undefined,
              "hideModal": [Function],
              "isRequestMade": false,
              "onSubmit": [Function],
              "open": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isRequestMade": false,
              "message": "Are you sure you want to delete this Event",
              "onCancel": [Function],
              "onConfirm": [Function],
              "open": false,
              "title": "Delete Event",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
